src/sdformat_urdf_plugin.cpp

```diff
@@ -0,0 +1,58 @@
+// Copyright 2020 Open Source Robotics Foundation, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+
+#include <rcutils/logging_macros.h>
+#include "sdformat_urdf/sdformat_urdf.hpp"
+#include <urdf_parser_plugin/parser.h>
+
+namespace sdformat_urdf {
+
+class SDFormatURDFParser : public urdf::URDFParser
+{
+public:
+
+  SDFormatURDFParser() = default;
+  virtual ~SDFormatURDFParser() = default;
+
+  urdf::ModelInterfaceSharedPtr
+  parse(const std::string & data) override
+  {
+    urdf::ModelInterfaceSharedPtr urdf_cpp;
+    sdf::Errors errors;
+    urdf_cpp = sdformat_urdf::parse(data, errors);
+
+    if (urdf_cpp) {
+      return urdf_cpp;
+    }
+
+    for (const sdf::Error & error : errors) {
+      RCUTILS_LOG_ERROR_NAMED("sdformat_urdf", "%s", error.Message().c_str());
+    }
+    if (errors.empty()) {
+      RCUTILS_LOG_ERROR_NAMED("sdformat_urdf", "Failed to parse but no errors reported");
+    }
+    return nullptr;
+  }
+
+  size_t
+  might_handle(const std::string & data) override
+  {
+    return data.find("<sdf");
```

Comment:
As with the URDF plugin, I think this is a bit too simplistic.  I'd recommend doing the same thing as we did over there, using tinyxml2 to do the parsing to check for the root element.

Reply:
Use tinyxml2 to filter valid XML documents that are not SDFormat in 7a5ad72442456120647b863d9522085c4c5284ff